src/components/TablePagination/TablePagination.tsx

```diff
@@ -0,0 +1,181 @@
+import React, {
+  ChangeEvent,
+  Children,
+  HTMLAttributes,
+  PropsWithChildren,
+  ReactElement,
+  ReactNode,
+  RefObject,
+  cloneElement,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import classnames from "classnames";
+import { usePagination } from "hooks";
+import Select from "components/Select";
+import TablePaginationControls from "./TablePaginationControls";
+import "./TablePagination.scss";
+
+/**
+ * Determine if we are working with a small screen.
+ * 'small screen' in this case is relative to the width of the description div
+ */
+const figureSmallScreen = (descriptionRef: RefObject<HTMLDivElement>) => {
+  const descriptionElement = descriptionRef.current;
+  if (!descriptionElement) {
+    return true;
+  }
+  return descriptionElement.getBoundingClientRect().width < 230;
+};
+
+/**
+ * Iterate direct react child components and override the value of the prop specified by @param dataForwardProp
+ * for those child components.
+ * @param children - react node children to iterate
+ * @param dataForwardProp - the name of the prop from the children components to override
+ * @param data - actual data to be passed to the prop specified by @param dataForwardProp
+ */
+const renderChildren = (
+  children: ReactNode,
+  dataForwardProp: string,
+  data: unknown[]
+) => {
+  return Children.map(children, (child) => {
+    return cloneElement(child as ReactElement, {
+      [dataForwardProp]: data,
+    });
+  });
+};
+
+const DEFAULT_PAGE_LIMITS = [50, 100, 200];
+const generatePagingOptions = (pageLimits: number[]) => {
+  return pageLimits.map((limit) => ({ value: limit, label: `${limit}/page` }));
+};
+
+export type Props = PropsWithChildren<{
+  /**
+   * list of data elements to be paginated. This component is un-opinionated about
+   * the structure of the data but it should be identical to the data structure
+   * reuiqred by the child table component
+   */
+  data: unknown[];
+  /**
+   * this is the prop of the underlying table component that should be overwritten with
+   * paginated data. If now specified, the default value for this prop is set to @constant rows
+   * which is the data prop for the @func MainTable component
+   */
+  dataForwardProp?: string;
+  /**
+   * the name of the item associated to each row within the table.
+   */
+  itemName?: string;
+  /**
+   * custom styling for the pagination container
+   */
+  className?: string;
+  /**
+   * custom description to be displayed by the pagination
+   */
+  description?: ReactNode;
+  /**
+   * custom per page limits express as an array of numbers. default is [50, 100, 200]
+   */
+  pageLimits?: number[];
+  /**
+   * render the pagination component below the table? default to "above"
```

Comment:
Better not mention defaults in the docs to keep a single source of truth in the code.
```suggestion
   * place the pagination component above or below the table?
```